### DISCUSSION TITLE
Feature Request: Flux language - Visual Query Builder

### DISCUSSION THREAD

**Discussion by rjvtww** [grafana/grafana#43848] [Observability]
**What would you like to be added**:

Add visual query builder functionality for the Flux query language on InfluxDB v2

**Why is this needed**:

For my organization, the visual query builder is everything. It enables any form of user to easily query data. Without it, I cannot start exploring a transition into the flux query language, together with all the nice features it allows inside of influxdb

**Comment by dprokop** [comment]
@ryantxu is this something on the Edge squad agenda?

**Comment by johnatannvmd** [comment]
IMHO you should push the Influx Data team for this, not Grafana Team.

OFFTOPIC: If some company releases a new great product and wants to integrate it with Grafana who should publish the plugin?

**Comment by ivanahuckova** [comment]
> If some company releases a new great product and wants to integrate it with Grafana who should publish the plugin?

@marcusolsson

**Comment by rjvtww** [comment]
Would be fantastic to hear from the Grafana team if this is in the plans, and if so, a rough idea of when it can be expected!

**Comment by marcusolsson** [comment]
> If some company releases a new great product and wants to integrate it with Grafana who should publish the plugin?

The company would let us (Grafana Labs) know, and we'd publish it on grafana.com/plugins. 

- Plugins for open source project are reviewed and published for free
- Plugins for commercial offerings require a paid plugin subscription

To submit your plugin to be published, you create a PR to the [plugin repository](https://github.com/grafana/grafana-plugin-repository/).

**Comment by andyl** [comment]
No query builder for Flux - OMG what a giant step backwards.  How could Influx Data let this happen??

**Comment by belperite** [comment]
My company would like people with little/no development or database experience to be able to pick from the available data items to create graphs, so this is a must-have for us too. Can work around it in the meantime by building the query in the InfluxDB UI, but the copy-pasting the queries to the Grafana UI isn't ideal.

**Comment by houmoller** [comment]
we to rely heavily on having non-technical people design dashboards. This is no longer possible using Flux.

**Comment by wfjm** [comment]
+1.  
Flux is powerful, but comes with a steep learning curve.  
I don't care who implements a Flux query editor, but both Influx and Grafana should have an interest.
Given that queries are much more compact in InfluxQL we'll keep using it even in InfluxV2 installations.  
Only when Flux is really needed it will be used via a second data source.   
Without a Flux query editor the transition from InfluxQL to Flux will be slow, if it happens at all.

**Comment by gabor** [comment]
we are moving feature-requests from github-issues to github-discussions.

**Comment by anwu** [comment]
Please?

**Comment by PeterMilovcik** [comment]
With the InfluxDB 2+, not supported InfluxQL for it, and no Visual Query Builder for Flux alternative, it is a major blocker for our usage of Grafana itself.

**Reply by johnatannvmd** [reply]
what are Grafana alternatives?

**Reply by nward** [reply]
> what are Grafana alternatives?

Influx ships a competent GUI comparable to Grafana and has a reasonable query builder which allows for exploration of the schema.

**Comment by vdwpsmt** [comment]
Flux is way more powerful then influxQl and opens up tons of extra possibilities in terms of loading and transforming data or even joining influxdata with data from sql,API's, csv files, etc....  
But it comes with one major drawback and that is indeed the steep learning curve.  So in order to convince the majority of people to make the switch, a minimum requirement is a query builder.

**Comment by benganellison** [comment]
I think the influx team has little to no interest in this, they want us to use influxdb 2 UI to build the query and force us into using their full tool chain. But grafana has a big interest in keeping us using the tool we have got used to

**Reply by vdwpsmt** [reply]
If influx wants us to use their UI then they will need to work on the user management of it.  No roles other than owner for now (cloud version).  This means that you need to give full permissions to anyone who wants to do (readonly) queries only.   deleting a complete bucket is a 2 step simple button clicking action.  This is a no-go for any company who is serious about data collection and security.

As for the alternatives: the flux plugin for vscode (https://docs.influxdata.com/influxdb/cloud/tools/flux-vscode/) seems to be the best alternative for now, as far as I can see, it is only a few steps away from a full query builder functionality.

**Comment by ajsicek** [comment]
does anybody know, what is actually the situation with this?
is that on a plan someday or has it been decided, it will never happen?
or?

**Comment by vdwpsmt** [comment]
Influx is spending a lot of effort in it's new InfluxDB (IOx).  This has native SQL query language support.  So not a lot of hope that a lot of extra effort on flux support will happen in the near future.  At least not by people of Influx itself.

**Reply by ajsicek** [reply]
oh thanks, that means grafana team is not working on it at all? is that usual approach to all datasources, that someone else does it?

**Comment by mcdaigle1** [comment]
I agree with the comments above.  The ability to browse resources and build queries was one of the major draws of Grafana.  I've set up Grafana multiple times at past companies, but haven't touched it an a few years.  Now I see they've actually removed useful functionality (the query builder).  I'm not sure I can recommend Grafana at my current company, as non-technical users will have a lot of difficulty using it.

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was converted into an issue, which is available here: https://github.com/grafana/grafana/issues/73432
